rules-tests/Php80/Rector/Class_/ClassPropertyAssignToConstructorPromotionRector/Fixture/mirror_additional_docblock.php.inc

```diff
@@ -0,0 +1,44 @@
+<?php
+
+namespace Rector\Tests\Php80\Rector\Class_\ClassPropertyAssignToConstructorPromotionRector\Fixture;
+
+use Doctrine\ORM\Mapping\Id;
+
+class MirrorAdditionalDocblock
+{
+    /**
+     * @var string
+     * @Id
+     */
+    public $name;
+
+    public function __construct($name)
+    {
+        $this->name   = $name;
+    }
+}
+
+?>
+-----
+<?php
+
+namespace Rector\Tests\Php80\Rector\Class_\ClassPropertyAssignToConstructorPromotionRector\Fixture;
+
+use Doctrine\ORM\Mapping\Id;
+
+class MirrorAdditionalDocblock
+{
+    /**
+     * @param string $name
+     */
+    public function __construct(
+        /**
+         * @Id
```

Comment:
I'd expect there to be either `@var string` here, or `public string $name` below :thinking:

Reply:
No, the `@var` already moved to `@param` to fill the param doc type.

If the `property` is `non-typed`, and the `param` is `non-typed` as well, this rule is not changing to typed.

Reply:
I see, that's the role of another rule I must be using 👍

Reply:
You can use `TypedPropertyRector` with set `INLINE_PUBLIC` to true to make property typed for public property:

```php
use Rector\Php74\Rector\Property\TypedPropertyRector;
// ...
    $rectorConfig
        ->ruleWithConfiguration(TypedPropertyRector::class, [
            // Set to false if you use in libraries, or it does create breaking changes.
            TypedPropertyRector::INLINE_PUBLIC => true,
        ]);
```